### PR TITLE
Introduced rotation methods for particles and galaxies

### LINKED ISSUE #12: Need to include geometry change helper functions
Where should we put these? Things like rotation matrices, face-on, edge-on. Possibly in the galaxy class?

### PR DESCRIPTION
**Pull request by WillJRoper**
This PR introduces the ability to rotate a galaxy/component either in place or returning a new instance. An example demonstrating it has been introduced. 

At the galaxy level the user can select which component's angular momentum should be used for the vector to rotate towards. 

To make this work an angular momentum property has been introduced on a `Particles` object.

Close #12 (finally).

## Issue Type

- Enhancement

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by WillJRoper** [comment]
rot_example

This is the image the example produces demonstrating rotating to face on and edge on based on stellar and gas angular momentum

File: src/synthesizer/utils/geometry.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    unit = np.array([0.0, 1.0, 0.0])
+
+    # Get the rotation matrix
+    matrix = get_rotation_matrix(vector, unit)
+"""
+
+import numpy as np
+
+
+def get_rotation_matrix(vector, unit=None):
```
Comment (aswinpvijayan) [inline src/synthesizer/utils/geometry.py:21]: The method is taken directly from stackoverflow, would be nice to quote that: https://stackoverflow.com/questions/43507491/imprecision-with-rotation-matrix-to-align-a-vector-to-an-axis

Comment (WillJRoper) [reply src/synthesizer/utils/geometry.py:21]: Included

**Comment by WillJRoper** [comment]
There has been a significant ruff update that meant everything was breaking (they have improved the import sorting from what I can tell, it now appears to automatically separate internal and external imports).

I've updated the workflows and pre-commit hook to account for this but it means there's a ton of formatting.

**Review by aswinpvijayan — APPROVED** [review]

**Pull request merged**